lib/rubocop/cop/rspec/unspecified_exception.rb

```diff
@@ -0,0 +1,82 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for a specified error in checking raised errors.
+      #
+      # Enforces one of an Exception type, a string, or a regular
+      # expression to match against the exception message as a parameter
+      # to `raise_error`
+      #
+      # @example
+      #
+      #     # bad
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error
+      #
+      #     # good
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) { |err| expect(err.data).to eq(42) }
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) do |err|
+      #       expect(err.data).to eq(42)
+      #     end
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error('error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, 'error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(/err/)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, /err/)
+      #
+      #     expect { do_something }.not_to raise_error
+      class UnspecifiedException < Cop
+        MSG = 'Specify the exception being captured'.freeze
+
+        EMPTY_RAISE_ERROR_PATTERN = <<-PATTERN.freeze
+          (send
+            (block
+                (send nil? :expect) ...)
+            :to
+            (send nil? :raise_error)
+          )
+        PATTERN
+        def_node_matcher :empty_raise_error_call, EMPTY_RAISE_ERROR_PATTERN
+
+        def_node_matcher :raise_error_with_block, <<-PATTERN.freeze
```

Comment:
IMO this matcher makes code harder to understand. I would replace it with a simple method that checks whether the parent node is a block with an argument, like this one:

```ruby
def on_send(node)
  return unless empty_raise_error_call(node) &&
    !block_with_args?(node.parent)
    # ... rest of the code
end

def block_with_args?(node)
  return unless node && node.block_type?
  node.arguments?
end
```

Reply:
That's great, much simpler - I appreciate it!